e2e/ui/Enterprise SSO/setup_link.spec.ts

```diff
@@ -0,0 +1,154 @@
+): Promise<Locator | string> {
+  const rows = await page.getByRole('row').all();
+  const setupLinks = await Promise.all(rows.map(async (el: Locator) => el));
+  let row: Locator | string = '';
+
+  for (const link of setupLinks) {
+    const el = await link.innerText();
+    const cellContents = el.split('\t');
+    if (
+      cellContents.length > 1 &&
+      cellContents[tenantProductIndexes[0]] === tenant &&
+      cellContents[tenantProductIndexes[1]] === product
+    ) {
+      row = link;
+    }
+  }
+
+  return row;
+}
+
+test.describe('Admin Portal Enterprise SSO SetupLink using generic SAML 2.0', () => {
+  test('should be able to create setup link and sso connection using generic SAML 2.0', async ({ page }) => {
+    const tenant = 'acme-setuplink-test.com';
+    const product = 'acme-setuplink-test';
+
+    // Go to admin/sso-connection/setup-link page and create setup link
+    await page.goto(TEST_SETUPLINK_ADMIN_URL);
+    await page.getByRole('button', { name: 'New Setup Link' }).click();
+    await page.getByPlaceholder('Acme SSO').fill('acme-test');
+    await page.getByLabel('Description (Optional)').fill('acme test');
+    await page.getByPlaceholder('acme', { exact: true }).fill(tenant);
+    await page.getByPlaceholder('MyApp').fill(product);
+    await page.getByPlaceholder('http://localhost:3366', { exact: true }).fill(TEST_SETUPLINK_REDIRECT_URL);
+    await page.getByPlaceholder('http://localhost:3366/login/').fill(TEST_SETUPLINK_DEFAULT_REDIRECT_URL);
+
+    await page.getByRole('button', { name: 'Create Setup Link' }).click();
+
+    // Extract generated setup link
+    const linkContent = await page
+      .getByText(TEST_SETUPLINK_URL_LABEL_SELECTOR)
+      .locator('input[type="text"]')
+      .first()
+      .inputValue();
+
+    // Go back to new connections page
+    await page.goto(TEST_SETUPLINK_ADMIN_URL);
+
+    // Await for rows loaded
+    await expect(page.getByRole('table')).toBeVisible();
+
+    // Get single row containing setuplink
+    let linkRow = await getRowByTenantProduct(page, tenant, product, [0, 1]);
+    expect(linkRow, 'Failed to create setup link').toBeTruthy();
+
+    // Open new tab and go to setup link page
+    const context = page.context();
+    const page1 = await context.newPage();
+    await page1.waitForLoadState();
+    await page1.goto(linkContent);
+
+    // Create SSO connection using generic SAML 2.0 workflow
+    await page1.getByRole('button', { name: 'Generic SAML 2.0' }).click();
+    await page1.getByRole('button', { name: 'Next Step' }).click();
+    await page1.getByRole('button', { name: 'Next Step' }).click();
+    await page1.getByPlaceholder('Paste the Metadata URL here').click();
+    await page1.getByPlaceholder('Paste the Metadata URL here').isVisible();
+    await page1.getByPlaceholder('Paste the Metadata URL here').fill(TEST_SETUPLINK_MOCK_METADATA_URL);
+    await page1.getByRole('button', { name: 'Save' }).click();
+
```

Comment:
Before closing page1 here, we can assert the added connection in the setup link sso listing.

```
await page1.waitForURL(/\/setup\/.+\/sso-connection$/);
await expect(page.getByRole('table')).toBeVisible();
// Assert connection is present in the table
```